core/kazoo_config/src/kapps_config_util.erl

```diff
@@ -80,3 +81,14 @@ account_schema(Config) when is_binary(Config) ->
     Name = account_schema_name(Config),
     {'ok', Schema} = kz_json_schema:load(Name),
     Schema.
+
+-spec system_config_document_schema(ne_binary()) -> kz_json:object().
+system_config_document_schema(Id) ->
+    Flat = [
```

Comment:
I think using the `[Key,...]` when constructing a `flat_object()` helps differentiate it from a normal `object()`. `{[{[<<"a">>, <<"b">>], <<"val">>}]}` is not a valid JSON object:

```erlang
> kz_json:encode({[{[<<"a">>, <<"b">>], <<"val">>}]}).
** exception throw: {error,{invalid_object_member_key,[<<"a">>,<<"b">>]}}
   in function  jiffy:encode/2 (src/jiffy.erl, line 93)
```

So I think, when constructing a `flat_object()`, we should opt to use `[Key]` over `Key`.

Thoughts?